mqtt_client.c

```diff
@@ -1030,7 +1013,8 @@ static esp_err_t deliver_publish(esp_mqtt_client_handle_t client)
                                           msg_total_len - msg_read_len > buf_len ? buf_len : msg_total_len - msg_read_len,
                                           client->config->network_timeout_ms);
         if (ret <= 0) {
-            return esp_mqtt_handle_transport_read_error(ret, client) == 0 ? ESP_OK : ESP_FAIL;
+            esp_mqtt_client_dispatch_transport_error(client);
```

Comment:
Please revert this change

Reply:
Reverted to call `esp_mtt_handle_transport_read_error(ret, client)` but then return ESP_FAIL unconditionally.